Everywhere: Some easy "DeprecatedFile" migrations

BenWiederhake opened the pull request:
This advances #17129 by doing I/O through `Core::File` instead of `Core::DeprecatedFile`, which gives us better OOM-handling for free, and (in the case of SystemServer) also should improve the ability to detect an extremely obscure bug (which apparently never occurs anyway).

Hendiadyoin1 commented on Userland/Libraries/LibJS/Runtime/VM.cpp:
```diff
@@ -938,15 +937,18 @@ ThrowCompletionOr<NonnullGCPtr<Module>> VM::resolve_imported_module(ScriptOrModu
-    StringView content_view { file_content.data(), file_content.size() };
+    auto file_content_or_error = file_or_error.value()->read_until_eof();
+    if (file_content_or_error.is_error()) {
+        return throw_completion<SyntaxError>(ErrorType::OutOfMemory, module_request.module_specifier);
```
Shouldn't it be an `InternalError` and not a syntax one?

BenWiederhake replied:
I have no idea! The argument in my head was that "insanely large file == syntax error". Yup, I'll change it to InternalError then.

BenWiederhake commented:
Changes since last push:
- Changed the JS module loading OOM from a "SyntaxError" to an "InternalError".

LucasChollet commented:
I already opened #18529 for LibJS's migration.

And to continue the discussion, I think that neither a `SyntaxError` nor an `InternalError` fit perfectly. OOM isn't the only error that can arise from `read_until_eof`.

BenWiederhake commented:
Changes since last push:
- Dropped the JS module commit entirely, in favor of #18529 (Thanks @LucasChollet!)

AtkinsSJ commented on Userland/Services/SystemServer/main.cpp:
```diff
@@ -72,18 +72,19 @@ static ErrorOr<void> determine_system_mode()
         // Continue and assume "text" mode.
         return {};
     }
+    const DeprecatedString system_mode = DeprecatedString::copy(system_mode_buf_or_error.value(), Chomp);
```
Nit: east const

BenWiederhake replied:
Huh, weird. Usually I prefer east const, but in this particular case I didn't even notice that there was a choice. Thanks!

AtkinsSJ commented on Userland/Services/SystemServer/main.cpp:
```diff
@@ -72,18 +72,19 @@ static ErrorOr<void> determine_system_mode()
-    auto f = Core::DeprecatedFile::construct("/sys/kernel/constants/system_mode");
-    if (!f->open(Core::OpenMode::ReadOnly)) {
-        dbgln("Failed to read system_mode: {}", f->error_string());
+    auto f_or_error = Core::File::open("/sys/kernel/constants/system_mode"sv, Core::File::OpenMode::Read);
```
Nit: I know it had the name `f` before, but can we call it `file_or_error` instead?

BenWiederhake commented:
Changes since last push:
- Instead of re-using the old variable name `f` unreflected, I changed it to `file` (Thanks @AtkinsSJ!)
- Rebased for convenience.
